code/modules/reagents/chemistry/reagents/alcohol_reagents.dm

```diff
@@ -77,6 +77,14 @@ All effects don't start immediately, but rather get worse over time; the rate is
 				// +10% success propability on each step, useful while operating in less-than-perfect conditions
 	return ..()
 
+/datum/reagent/consumable/ethanol/on_mob_add(mob/living/L,amount)
```

Comment:
Do you think this could be `on_mob_metabolize()` instead?

Reply:
went with on_mob_add so i could use amount there, though it is fully possible it's unnecessary for some reason or another

Reply:
You could use the `volume` var instead, but whatever.

Reply:
yeah i rethought it and made it less of a "meme effect", maybe recheck it